File: client/src/lib/invoke/saveFactureToFile.ts
import { invoke } from "@tauri-apps/api/core";
import { ResultAsync } from "neverthrow";
import type { Invoice } from "@bindings";

export function saveFactureToFile(
    invoice: Invoice,
    path_string: string
): ResultAsync<void, string> {
    return ResultAsync.fromPromise(
        invoke("save_facture_to_file", { invoice, path_string }),
        (error) => `Failed to save facture to file: ${error}`
    );
}

File: client/src-tauri/src/commands/save_facture_to_file.rs
use crate::core::invoice::Invoice;

#[tauri::command]
pub async fn save_facture_to_file(
    invoice: Invoice,
    path_string: String,
) -> Result<(), String> {
    tauri::async_runtime::spawn_blocking(move || {
        Invoice::to_file(&invoice, &path_string).map_err(|err| err.to_string())
    })
    .await
    .map_err(|err| format!("Failed to save facture to file: {}", err))?
}
